### DISCUSSION TITLE
Wrap useAsyncData to another function

### DISCUSSION THREAD

**Discussion by NisuSan** [nuxt/nuxt#19885] [Questions]
Hi. I faced with the problem when wrap ```useAsyncData``` to another function, the **watch** property does not work any more

```javascript
// functions.ts
export async function testAsync(path: string, watch: (WatchSource<unknown> | object)[]) {   
  return await useAsyncData('filters', () => $fetch(`/api/${path}`, { method: 'GET' }), { watch })
}
```

And call it in another component

```javascript
const props = defineProps<{modelValue: ToolBarModelValue}>()
let { data: filters } = await testAsync(groupsToApi[props.modelValue.group], [() => props.modelValue.group])
```

When use pure ```useAsyncData``` in my component, watch property works as expected. Have someone faced with this issue?

**Comment by telisjok** [comment]
Why noone has given an answer to this yet? This would be a good way of caching api calls.
